### PR TITLE
fix: fix transaction handling + use async everywhere

### PR DESCRIPTION
**Pull request by alexgarel**
* fix transaction handling (move away from auto-commit, and encapsulate http calls in a transaction)
* use aiopg to keep things async
* fix usage of requests to use aiohttp (async)
* change tests to be more accurate (but destructive)
* run tests on pull request

### CODE REVIEW

File: local_settings_example.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
-# OpenFoodFacts username and password for authentication
-USER = 'myusername'
-PASSWORD = 'mypassword'
+# OpenFoodFacts username and password for authentication for tests only !
```
Comment (raphael0202) [inline local_settings_example.py:1]: Very minor comment, but there is never a space before '!' in English

File: folksonomy/api.py [hunk 2]
```diff
@@ -195,6 +197,19 @@ async def authentication(response: Response, session: Optional[str] = Cookie(Non
...
+        params.append(k)
+        if v != '':
+            conditions.append('v=%s')
+            params.append(v)
+    where = " AND ".join(conditions)
+    return where, params
+
+
 @app.get("/products/stats", response_model=List[ProductStats])
 async def product_stats(response: Response,
```
Comment (raphael0202) [inline folksonomy/api.py:214]: It can be done in another PR, but it would be nice to have black formatting

File: folksonomy/api.py [hunk 3]
```diff
@@ -195,6 +197,19 @@ async def authentication(response: Response, session: Optional[str] = Cookie(Non
         status_code=500, detail="Server error")
 
 
+def property_where(owner, k, v):
```
Comment (raphael0202) [inline folksonomy/api.py:200]: A docstring and parameter typing would be nice here!

**Review by raphael0202 — APPROVED** [review]

File: folksonomy/api.py [hunk 4]
```diff
@@ -64,39 +54,43 @@ async def startup_event():
...
 async def get_current_user(token: str = Depends(oauth2_scheme)):
     """
     Get current user and check token validity if present
     """
     if token and '__U' in token:
-        query = cur.mogrify(
-            "UPDATE auth SET last_use = current_timestamp AT TIME ZONE 'GMT' WHERE token = %s", (token,))
-        cur.execute(query)
+        cur = db.cursor()
+        await cur.execute(
```
[the remark this answers is not in the document]
Comment (alexgarel) [reply folksonomy/api.py:80]: Yes, but I think this was an opinionated choice (to be able to fully use postgresql capabilities).

File: folksonomy/api.py (hunk 2 above)
Comment (alexgarel) [reply folksonomy/api.py:214]: I agree but it's better to have a single commit of reformatting.

**Pull request merged**